services/user/XAdmin/ui/src/pages/create-page.tsx

```diff
@@ -99,13 +87,38 @@ export const CreatePage = () => {
         resolver: zodResolver(chainTypeSchema),
     });
 
-    const { canNext, canPrev, next, previous, currentStep, maxSteps } =
-        useStepper(4, [chainTypeForm, blockProducerForm, "a"]);
-
-    const { data: packages } = usePackages();
+    const keyDeviceForm = useForm<z.infer<typeof KeyDeviceSchema>>();
 
     const isDev = chainTypeForm.watch("type") == "dev";
     const bpName = blockProducerForm.watch("name");
+    const keyDevice = keyDeviceForm.watch("id");
+
+    const Step = {
+        ChainType: "CHAIN_TYPE",
+        BlockProducer: "BLOCK_PRODUCER",
+        KeyDevice: "KEY_DEVICE",
+        Confirmation: "CONFIRMATION",
+        Completion: "COMPLETION",
+    } as const;
+
+    type StepKey = (typeof Step)[keyof typeof Step];
+
+    const {
+        canNext,
+        canPrev,
+        next,
+        previous,
+        currentStepNum,
+        currentStep,
+        maxSteps,
+    } = useStepper<StepKey>([
+        { step: Step.ChainType, form: chainTypeForm },
+        { step: Step.BlockProducer, form: blockProducerForm },
+        { step: Step.KeyDevice, form: keyDeviceForm, skip: isDev },
+        { step: Step.Confirmation },
+    ]);
```

Comment:
Gives me `enum` vibes

```suggestion
    const Step = z.enum([
        "ChainType",
        "BlockProducer",
        "KeyDevice",
        "Confirmation",
        "Completion",
    ]);

    type StepKey = z.infer<typeof Step>;

    const {
        canNext,
        canPrev,
        next,
        previous,
        currentStepNum,
        currentStep,
        maxSteps,
    } = useStepper<StepKey>([
        { step: Step.Enum.ChainType, form: chainTypeForm },
        { step: Step.Enum.BlockProducer, form: blockProducerForm },
        { step: Step.Enum.KeyDevice, form: keyDeviceForm, skip: isDev },
        { step: Step.Enum.Confirmation },
    ]);
```

Reply:
Yes, it works similar to an enum, but better. I try to avoid TS enums, as they're considered harmful by many. As for zod here, I get the value add from use cases with external data sources like API responses or user inputs. But I'm hesitant to zodify everything. And while we iterate early on, you'll see me opting for plain TS for dev speed and simplicity.